vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java

```diff
@@ -693,10 +702,22 @@ public void setTarget(Component target) {
         target.getUI().ifPresent(this::onTargetAttach);
         targetAttachRegistration = target
                 .addAttachListener(e -> onTargetAttach(e.getUI()));
+        target.addDetachListener(e -> {
```

Comment:
There's also another case where:
- You set a target that is attached
- Popover gets auto-added
- You set a target that is not attached
- Popover stays in UI

Maybe setting a new target should always auto remove, and then it would then immediately be added again if the new target is attached. Could maybe also test that removing and adding the same popover in the same roundtrip doesn't recreate all the elements on the client-side, otherwise client-side state (scroll pos., focus) would get lost. Though that's maybe not a big deal when switching targets?

Reply:
> Looks like the detach listener is not removed when changing the target. Probably not a common use case, but maybe this should work in the same way like the attach registration above. 

Good catch, fixed.

Reply:
> Maybe setting a new target should always auto remove, and then it would then immediately be added again if the new target is attached. 

Thanks, makes sense. Changed the logic.

> Could maybe also test that removing and adding the same popover in the same roundtrip doesn't recreate all the elements on the client-side, otherwise client-side state (scroll pos., focus) would get lost.

Do you have suggestions how to check this? I'm not 100% sure this is doable with only unit test.

Reply:
Tested manually by setting `theme="arrow"` using dev tools and changing target - the attribute is still there after re-adding, so the element is not recreated.